src/plugin.js

```diff
@@ -138,6 +138,18 @@ module.exports = class PluginXrpEscrow extends EventEmitter2 {
     return fulfillment
   }
 
+  async submit (signed) {
+    const txHash = signed.id
+    const result = new Promise((resolve, reject) => {
+      this._submitted[txHash] = { resolve, reject }
```

Comment:
Might be a good idea to add some kind of time-out just in case the connection drops and a notification is missed. The time out event could even query the ledger for the tx state

Reply:
I'll make a follow-up ticket for that